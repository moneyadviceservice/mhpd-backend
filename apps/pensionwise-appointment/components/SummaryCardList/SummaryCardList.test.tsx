import React from 'react';
import { render, screen } from '@testing-library/react';
import { SummaryCardList } from '.';
import mockData from './SummaryCardList.mock.json';

jest.mock('next/router', () => ({
  useRouter: jest.fn().mockReturnValue({
    query: {},
  }),
}));

describe('SummaryCardList component', () => {
  it('renders correctly', () => {
    render(<SummaryCardList data={mockData} />);
    const container = screen.getByTestId('summary-card-list');
    expect(container).toMatchSnapshot();
  });
});
