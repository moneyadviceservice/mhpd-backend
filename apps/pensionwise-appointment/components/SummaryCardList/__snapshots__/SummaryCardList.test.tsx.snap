// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryCardList component renders correctly 1`] = `
<ul
  class="m-0 md:grid md:grid-cols-2 md:gap-6 xl:gap-y-10 md:auto-rows-fr"
  data-testid="summary-card-list"
>
  <li
    class="p-0 pb-6 md:p-0 m-0 md:flex"
    data-testid="summary-card-0"
  >
    <div
      class="t-information-callout border border-slate-400 rounded-bl-3xl w-full md:flex md:flex-col md:justify-between px-4 md:px-7 py-6"
      data-testid="information-callout"
    >
      <div>
        <h4
          class="md:text-2xl text-gray-800 font-bold text-xl mb-4"
          data-testid="summary-card-title"
        >
          Generate a summary
        </h4>
        <p
          class="mb-4"
          data-testid="paragraph"
        >
          Generate a summary of the pension options to refer back to.
        </p>
      </div>
      <p
        class="text-right text-sm"
        data-testid="summary-card-duration"
      >
        (
        approximately
         
        <span
          class="font-bold"
        >
          2
           
          minutes
          )
        </span>
      </p>
    </div>
  </li>
  <li
    class="p-0 pb-6 md:p-0 m-0 md:flex"
    data-testid="summary-card-1"
  >
    <div
      class="t-information-callout border border-slate-400 rounded-bl-3xl w-full md:flex md:flex-col md:justify-between px-4 md:px-7 py-6"
      data-testid="information-callout"
    >
      <div>
        <h4
          class="md:text-2xl text-gray-800 font-bold text-xl mb-4"
          data-testid="summary-card-title"
        >
          Update your beneficiary
        </h4>
        <p
          class="mb-4"
          data-testid="paragraph"
        >
          Update your beneficiary
        </p>
      </div>
      <p
        class="text-right text-sm"
        data-testid="summary-card-duration"
      >
        (
        approximately
         
        <span
          class="font-bold"
        >
          15
           
          minutes
          )
        </span>
      </p>
    </div>
  </li>
  <li
    class="p-0 pb-6 md:p-0 m-0 md:flex"
    data-testid="summary-card-2"
  >
    <div
      class="t-information-callout border border-slate-400 rounded-bl-3xl w-full md:flex md:flex-col md:justify-between px-4 md:px-7 py-6"
      data-testid="information-callout"
    >
      <div>
        <h4
          class="md:text-2xl text-gray-800 font-bold text-xl mb-4"
          data-testid="summary-card-title"
        >
          Get regulated advice
        </h4>
        <p
          class="mb-4"
          data-testid="paragraph"
        >
          Get some regulated advice
        </p>
      </div>
      <p
        class="text-right text-sm"
        data-testid="summary-card-duration"
      >
        (
        approximately
         
        <span
          class="font-bold"
        >
          2
           
          minutes
          )
        </span>
      </p>
    </div>
  </li>
</ul>
`;
